File: sdk/python/examples/simple_agent.py
"""最简示例 Agent — 基于动量策略预测 BTC 15 分钟走势"""

import os
import logging
from hive_protocol import HiveAgent, Prediction
from hive_protocol.types import PredictionResult, RoundInfo, RoundResult

logging.basicConfig(level=logging.INFO, format="%(asctime)s [%(name)s] %(message)s")

agent = HiveAgent(
    rpc_url=os.environ.get("RPC_URL", "https://mainnet-rpc.axonchain.ai/"),
    private_key=os.environ["AGENT_PRIVATE_KEY"],
    contract_addresses={
        "round": os.environ["HIVE_ROUND_ADDRESS"],
        "agent": os.environ["HIVE_AGENT_ADDRESS"],
        "vault": os.environ["HIVE_VAULT_ADDRESS"],
        "score": os.environ["HIVE_SCORE_ADDRESS"],
    },
)

# 先注册（如果已注册会自动跳过）
agent.register(stake_axon=100)


@agent.on_new_round
def predict(info: RoundInfo) -> PredictionResult:
    """
    简单动量策略：
    - 价格在涨 → 预测继续涨
    - 价格在跌 → 预测继续跌
    - 信心度基于波动幅度
    """
    trend = info.market_snapshot.get("15m_trend", 0)

    if isinstance(trend, str):
        trend = float(trend.replace("%", "")) / 100

    if trend > 0:
        direction = Prediction.UP
    else:
        direction = Prediction.DOWN

    confidence = min(int(abs(trend) * 10000) + 30, 70)

    return PredictionResult(direction=direction, confidence=confidence)


@agent.on_round_result
def on_result(result: RoundResult):
    status = "✅" if result.correct else "❌"
    print(f"Round {result.round_id}: {status} | "
          f"PnL: {result.profit_loss:+.2f} U | "
          f"Score: {result.new_hive_score}")


if __name__ == "__main__":
    agent.start()

File: sdk/typescript/examples/simple-agent.ts
import { HiveAgent, Prediction, type PredictionResult, type RoundInfo } from "../src";

const agent = new HiveAgent({
  rpcUrl: process.env.RPC_URL || "https://mainnet-rpc.axonchain.ai/",
  privateKey: process.env.AGENT_PRIVATE_KEY as `0x${string}`,
  contractAddresses: {
    round: process.env.HIVE_ROUND_ADDRESS as `0x${string}`,
    agent: process.env.HIVE_AGENT_ADDRESS as `0x${string}`,
    vault: process.env.HIVE_VAULT_ADDRESS as `0x${string}`,
    score: process.env.HIVE_SCORE_ADDRESS as `0x${string}`,
  },
});

agent.onNewRound(async (round: RoundInfo): Promise<PredictionResult> => {
  // 简单动量策略
  const trend = (round.marketSnapshot["15m_trend"] as number) || 0;

  return {
    direction: trend > 0 ? Prediction.UP : Prediction.DOWN,
    confidence: Math.min(Math.abs(trend * 10000) + 30, 70),
  };
});

agent.onRoundResult(async (result) => {
  const icon = result.correct ? "✅" : "❌";
  console.log(
    `Round ${result.roundId}: ${icon} | PnL: ${result.profitLoss > 0 ? "+" : ""}$${result.profitLoss} | Score: ${result.newHiveScore}`
  );
});

async function main() {
  await agent.register();
  await agent.start();
}

main().catch(console.error);
